### ISSUE TITLE
Expose consistent client method set

### ISSUE THREAD

**Issue by eapache** [Shopify/sarama#23]
The `Client` object has basically no public methods right now, it is just a place to dump methods (mostly broker and metadata tracking) that are shared by the Consumer and the Producer. Its primary functions should have specific semantics defined and then be published so other people can make use of them.

Probably depends on getting #15 right first...

I'm thinking (as a rough first draft):
 * `Leader(topic, partition)` returns the broker object leading the topic/partition
 * `Topics()` returns the list of topics for the cluster
 * `Partitions(topic)` returns the list of partitions for the topic
 * `RefreshMetadata(topics[])`

Open Questions:
 * Does it make sense to store/return other metadata (replicas or ISRs, for example)? Does any of this provide benefit beyond just sending a MetadataRequest using the low-level Broker object?
 * How are brokers returned to the client when no longer needed? (the current `disconnectBroker` feels hacky, this really depends on #15)
 * Do we want to provide `RefreshTopic(topic)` or `RefreshAllTopics()` helper functions or something like that?
 * Does it make sense to expose `any()`? It's only useful for metadata, so this probably depends if we expose all metadata or if a user might want to get it themselves.

**Comment by eapache** [comment]
All of the methods I started this ticket for have been created (a while ago, in most cases). The only open question is broker disconnection, which is still better-tracked by #15. This can be closed.
